core/app/modules/import-export/directories/base.php

```diff
@@ -50,6 +55,22 @@ public function __construct( Iterator $iterator, Base $parent = null ) {
 		$this->parent = $parent;
 
 		$this->register_directories();
+
+		add_filter( 'elementor/document/save/data', [ $this, 'prevent_saving_elements_on_post_creation' ], 10, 2 );
+	}
+
+	public function prevent_saving_elements_on_post_creation( $data, $document ) {
+		if ( $data['elements'] ) {
+			$this->set_documents_elements_by_id( $document->get_main_id(), $data['elements'] );
+
+			$data['elements'] = [];
+		}
+
+		return $data;
+	}
+
+	private function set_documents_elements_by_id( $id, $elements ) {
+		self::$documents_elements[ $id ] = $elements;
```

Comment:
`self` is shared with all extends of the base.
Can you replace across the file?
```suggestion
		static::$documents_elements[ $id ] = $elements;
```

Reply:
I need it to be shared with all extended classes.
The $this keyword in line 125 is different in each iteration.